viewing_party/main.py

```diff
@@ -0,0 +1,251 @@
+#WAVE ONE
+
+def create_movie(title, genre, rating):  
+    movie_dict = {} 
+    if title and genre and rating:
+        movie_dict["title"] = title
+        movie_dict["genre"] = genre
+        movie_dict["rating"] = rating
+        return movie_dict
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie) 
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    #This represents that the user has a watchlist and\
+    #a list of watched movies
+    for movie_dict in user_data["watchlist"]:
+        if movie_dict["title"] == title:
+            user_data["watchlist"].remove(movie_dict)
+            user_data["watched"].append(movie_dict)
+            return user_data
+    else:
+        return user_data
+
+#WAVE TWO
+
+def get_watched_avg_rating(user_data):
+    # the value of user_data will be a dictionary with a\
+    #a "watched" list of movie dictionaries
+    #This represents that the user has a list of watched\
+    #movies
+    avg_rating = 0.0
+    count =0
+    rating_sum = 0.0
+    for movie_dict in user_data["watched"]:
+        for key,value in movie_dict.items():
+            if key == "rating":
+                rating_sum += value
+                count += 1
+    #in place to settle division by zero error:
+    if count == 0:
+        avg_rating = 0
+    else:
+        avg_rating = rating_sum/count
+    return avg_rating
+
+def get_most_watched_genre(user_data):
+    genre_count_dic ={}
+    genre_count_list =[]
+    for movie_dict in user_data["watched"]:
+        for key,value in movie_dict.items():
+            if key == "genre":
+                genre_count_list.append(value)
+    for x in genre_count_list:
+        if not x in genre_count_dic:
+            genre_count_dic[x] =1
+        else:
+            genre_count_dic[x] +=1
+
+#obtain highest value in genre_count_dic 
+#seperated key & value in two seperate lists
+    k_list =[]
+    v_list =[]
+    for key,value in genre_count_dic.items():
+        k_list.append(key)
+        v_list.append(value)
+#incase of an empty list returning None
+    if len(k_list) == 0:
+        genre_most = None
+        return genre_most
+    else:
+        maximum = v_list[0]
+        for x in v_list:
+            if x > maximum:
+                maximum= x 
+#finding corresponding index of maximum value in key list
+        g= v_list.index(maximum)
+        genre_most = k_list[g]
+        return genre_most
+
+#WAVE THREE
+
+def get_unique_watched(user_data):
```

Comment:
This function works and that's great.

A few areas to improve in:

1.  You should use more descriptive variable names than x, y and z.
2. You have some extra looping through dictionaries to find specific keys.  You don't need to do that.